src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -340,12 +362,15 @@ def get_validate_platform(cmd, os_type, platform):
     if platform:
         platform_split = platform.split('/')
         platform_os = platform_split[0]
-        platform_arch = platform_split[1] if len(platform_split) > 1 else Architecture.amd64.value
-        platform_variant = platform_split[2] if len(platform_split) > 2 else None
+        platform_arch = platform_split[1] if len(
+            platform_split) > 1 else Architecture.amd64.value
+        platform_variant = platform_split[2] if len(
+            platform_split) > 2 else None
 
     if os_type and platform:
         if os_type.lower() != platform_os.lower():
-            raise CLIError("The OS in '--platform' should exactly match the value provided in '--os'.")
+            raise CLIError(
+                "The OS in '--platform' should exactly match the value provided in '--os'.")
```

Comment:
Why are these even allowed to be set independently then?

Reply:
//cc @shahzzzam

Reply:
we are deprecating `--os` in favor of `--platform`. Platform is also inclusive of the architecture. For instance, --os can only be [windows,linux] while `platform` can be [linux/arm, windows/amd64, amd64/v7]. 
But before deprecating, we want to make sure users can also do `--os windows --platform amd64`.

Reply:
Although  @jaysterp and I discussed this offline, and we believe that its better to error out when the user uses both `--os` and `--platform` together.

Reply:
see fix